library/blivet.py

```diff
@@ -791,29 +791,53 @@ def _get_params_lvmcache(self):
 
         return dict(cache_request=cache_request)
 
+    def _get_parent_thinpool_by_name(self, thin_pool_name):
+
+        if thin_pool_name is None:
+            if len(self._blivet_pool._thinpools) != 1:
+                raise BlivetAnsibleError("cannot determine thinpool for assigning volume '%s' " % self._volume.get('name'))
+            else:
+                return self._blivet_pool._thinpools[0]
```

Comment:
in the case where `len(self._blivet_pool._thinpools) == 1`, is it always true that `thin_pool_name == self._blivet_pool._thinpools[0].name`?

Reply:
Yes. BlivetLVMVolume._get_parent_thinpool_by_name is called from BlivetLVMVolume._create under condition that thin pool creation is requested.
But before BlivetLVMVolume._create is called BlivetLVMPool._manage_thin_pools is executed which creates thin pools (and/or finds the existing ones).